pkg/utils/kubernetes/crd_test.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package kubernetes_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsscheme "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/scheme"
+	cli "sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+var _ = Describe("CRD", func() {
+	var (
+		ctx context.Context
+
+		UnreadyCrd = &apiextensionsv1.CustomResourceDefinition{}
+		ReadyCrd   = &apiextensionsv1.CustomResourceDefinition{
+			Status: apiextensionsv1.CustomResourceDefinitionStatus{
+				Conditions: []apiextensionsv1.CustomResourceDefinitionCondition{
+					{Type: apiextensionsv1.Established, Status: apiextensionsv1.ConditionTrue},
+					{Type: apiextensionsv1.NamesAccepted, Status: apiextensionsv1.ConditionTrue},
+				},
+			},
+		}
+
+		ValidManifest = `apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+    name: myresources.mygroup.example.com`
+		InValidManifest = `thisIsNotAValidManifest`
+	)
```

Comment:
Can you assign those values in `BeforeEach`, to ensure each test has an initial state?